### PR TITLE
add Jenkinsfile

### PR DESCRIPTION
**Pull request by driesdeproost**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        stage("Clean") {
+            steps {
+                sh "./gradlew clean"
+            }
+        }
+
+        stage("Assemble") {
+            steps {
+                sh "./gradlew assemble --refresh-dependencies -i"
```
Comment (tgeens) [inline Jenkinsfile]: `--refresh-dependencies` makes it VERY slow and is used to solve issues, not intended for a standard build

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                    branch "release*"
+                }
+            }
+            environment {
+                SONATYPE_CREDENTIALS = credentials('sonatype')
+                GPGPASSPHRASE = credentials('gpgpassphrase')
+            }
+            steps {
+                script {
+                    sh "./gradlew publish -Pde_publish_username=${SONATYPE_CREDENTIALS_USR} -Pde_publish_password=${SONATYPE_CREDENTIALS_PSW} -PkeyId=DF8285F0 -Ppassword=${GPGPASSPHRASE} -PsecretKeyRingFile=/var/jenkins_home/secring.gpg"
```
Comment (tgeens) [inline Jenkinsfile]: `de_` ?

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        stage("Assemble") {
+            steps {
+                sh "./gradlew assemble --refresh-dependencies -i"
+            }
+        }
+
+        stage('Publish Amp') {
+            when {
+                anyOf {
+                    branch "master*"
```
Comment (tgeens) [inline Jenkinsfile]: it's OK to remove the branch constraints while working on this PR (in draft mode), untill you know the publishing actually works

Comment (driesdeproost) [reply Jenkinsfile]: ok, will reopen PR once publishing has succeeded

**Comment by driesdeproost** [comment]
closing as draft until publishing succeeded

**Comment by tgeens** [comment]
@driesdeproost You can put a PR in Draft-mode, with the "save" button on the bottom

**Comment by driesdeproost** [comment]
> @driesdeproost You can put a PR in Draft-mode, with the "save" button on the bottom

I was not aware of that
Cool, thanks

**Comment by driesdeproost** [comment]
seems you can only do that at creation

**Comment by driesdeproost** [comment]
Artifacts have been published

**Pull request closed without merging**